### PR TITLE
Update 'Homebrew/homebrew' references in comments/code/documentation

### PR DESCRIPTION
**Pull request by UniqMartin**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? ➡️ **N/A**
- [x] Have you successfully ran `brew tests` with your changes locally?

----

There are still a few places in our code base that reference our old repository. Replace `homebrew` with `brew`, `homebrew-core`, or `legacy-homebrew` depending on context.

### CODE REVIEW

File: Library/Homebrew/cmd/test-bot.rb [hunk 1]
```diff
@@ -336,7 +336,7 @@ def diff_formulae(start_revision, end_revision, path, filter)
         @name = @hash
       # Handle a URL being passed on the command-line or through Jenkins/Travis
       # environment variables e.g.
-      # `brew test-bot https://github.com/Homebrew/homebrew/pull/44293`.
+      # `brew test-bot https://github.com/Homebrew/legacy-homebrew/pull/44293`.
```
Comment (xu-cheng) [inline Library/Homebrew/cmd/test-bot.rb]: This should be updated to a url in homebrew/core as it's just an example.

Comment (UniqMartin) [inline Library/Homebrew/cmd/test-bot.rb]: I agree. Fixed it!

**Comment by mikemcquaid** [comment]
👍

**Comment by DomT4** [comment]
LGTM.

**Pull request closed without merging**